src/data_gathering.py

```diff
@@ -188,18 +202,24 @@ def get_non_mev_transactions():
     if not compare_and_validate_sql(confirmed_transactions_per_block, local_non_mev_query_sql):
         return None
 
-    # Get start and end blocks
-    latest_block_number = web3.eth.block_number
-    block_delay_seconds = config.get('block_delay_seconds', 0)
-    
-    # Adjusting the latest block number to account for block propagation delay
-    end_block = latest_block_number - int(block_delay_seconds // 12)  # Approximate block time for Ethereum is 12 seconds
-    latest_processed_block = get_latest_processed_block()
-    if latest_processed_block is not None:
-        start_block = latest_processed_block - config.get('start_block_offset', 100)
+    # Determine block range
+    if fulfill_gap:
+        # Use manually specified blocks
+        start_block = manual_start_block
+        end_block = manual_end_block
     else:
-        start_block = latest_block_number - config.get('start_block_offset', 100)
-    print(start_block, end_block)
+        # Get start and end blocks
+        latest_block_number = web3.eth.block_number
+        block_delay_seconds = config.get('block_delay_seconds', 0)
```

Comment:
I think we should just assume the largest block number -1, we find in the Dune table is the latest block.

Reply:
This logic was made Specially as it is, because Chen wanted it to be configurable (I mean, the delay).

The fulfilling gaps logic was not mentioned before, but is a good point to make it, so I decided to combine it with what we already have.
That's exactly why "fulfill_gap" Enables or disables the logic at all. In case of true, it bypasses the delay and offset (as logically, we need an exact gap).

Extract the latest block from dune, not from RPC (to be sure that is synced with Dune), that's a good point.

Reply:
How would someone want to run this script in the future? I think you would either want to run it for a specific block range or you would want to run it continuously (e.g. start off where you last were and index all new data available).

Having the concept of a `block_delay` is useful in the second case when we want to account for the time Dune takes to index all relevant transactions.

However, I'm also worried about the knowledge needed to correctly run/configure the script (5 values are too much).

If it's not possible to infer the most recent block dune has (by taking max block number - 1 from the `mevblocker.raw_bundles` table), I think we should at least reduce it to three

1.  start block (if not set, check if there is a last block stored in /data otherwise current block - 100)
2. end block (if not set, current block - block delay)
3. block delay